src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/AnalyzedArguments.cs

```diff
@@ -42,18 +39,23 @@ public BoundExpression Argument(int i)
             return Arguments[i];
         }
 
-        public string Name(int i)
+        public void AddName(IdentifierNameSyntax name)
+        {
+            Names.Add((name.Identifier.ValueText, name.Location));
+        }
+
+        public string? Name(int i)
         {
             if (Names.Count == 0)
             {
                 return null;
             }
 
-            IdentifierNameSyntax syntax = Names[i];
-            return syntax == null ? null : syntax.Identifier.ValueText;
+            var nameAndLocation = Names[i];
+            return nameAndLocation is (string name, _) ? name : null;
```

Comment:
>nameAndLocation is (string name, _) ? name : null [](http://example.com/codeflow?start=19&length=49)

```nameAndLocation?.Name``` feels much simpler and easier to follow.

Reply:
It absolutely is, sometimes you get attracted by the new shiny and forget the existing thing that works better.